TCD-2905: Coroner component and Morgue page (#2262)
saucelabs/sauce-docs: docs/error-reporting/advanced/coroner/client-getting-started.md

---
id: client-getting-started
title: Coroner Client - Getting Started
sidebar_label: Client Getting Started
description: Submit snapshot files (.btt files) to the object store (coronerd).
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import useBaseUrl from '@docusaurus/useBaseUrl';

> The `coroner` client tool allows you to submit snapshot files (.btt files) to the object store (coronerd). If you want to perform command-line querying of the Backtrace object store, you should use the [morgue tool](https://documentation.backtrace.io/product_product_morgue/) instead.

Author: ```suggestion
The `coroner` client tool enables you to submit snapshot files (.btt files) to the object store (coronerd). To perform command-line querying of the Backtrace object store, use the [morgue tool](https://documentation.backtrace.io/product_product_morgue/) instead.
```

For server-side configuration with CA-trusted certificates, refer to: [Database Install with CA-Trusted Certificates](https://documentation.backtrace.io/coronerd_install#ca-trusted-certificates).

```bash
[universe]
name = myuniversename
write = https://:6098
read = :4097
read.ssl.enabled = true
read.ssl.method = TLSv12

# Below, you will associate the tokens you created in the coroner console (token_id)
# with a concise name (token_name). You will use these token names when submitting
# snapshots to coroner later on.
[token]
=
```

### Configuration Using Self-Signed Certificates

For server-side configuration with self-signed certificates, refer to: [Database Install with Self-Signed Certificates](https://documentation.backtrace.io/coronerd_install#self-signed-certificates).

```bash
[universe]
name = myuniversename
write = https://:6098
read = :4097
read.ssl.enabled = true
read.ssl.method = TLSv12
curl.ssl_verifyhost = false
curl.ssl_verifypeer = false

# Below, you will associate the tokens you created in the coroner console (token_id)
# with a concise name (token_name). You will use these token names when submitting
# snapshots to coroner later on.
[token]
=
```

> ### Configuration For Unsecured Communications

Author: ```suggestion
### Configuration for Unsecured Communications
```

For server-side configuration with no SSL/TLS, refer to: [Database Install with No SSL/TLS](https://documentation.backtrace.io/coronerd_install#no-ssltls).

```bash
[universe]
name = myuniversename
write = http://:6097
read = :4097

# Below, you will associate the tokens you created in the coroner console (token_id)
# with a concise name (token_name). You will use these token names when submitting
# snapshots to coroner later on.
[token]
=
```

### Uploading Individual Snapshots with `coroner put`

The most common use for `coroner` is to submit individual snapshots to the object store. This is how the `coresnap` utility uploads the snapshots it creates.

To upload a snapshot, use the following command:

```bash
$ coroner put projectname tokenname /path/to/snapshot.btt
```

You can optionally include the `-c` flag to specify a coroner configuration file, and the `-u` flag to unlink the snapshot file when the upload is complete.

### Uploading Snapshots Automatically with `coroner daemon`

> The `coroner daemon` mode allows coroner to continuously monitor a list of folders for snapshot files. Whenever a snapshot is saved to one of these folders, `coroner` will automatically upload it to the object store using a specified token and then remove it. This is useful if you have a process that saves snapshots to a central location and you want to automatically upload them.

Author: ```suggestion
The `coroner daemon` mode allows coroner to monitor a list of folders for snapshot files continuously. Whenever a snapshot is saved to one of these folders, `coroner` will automatically upload it to the object store using a specified token and then remove it. It helps if you have a process that saves snapshots to a central location and you want to upload them automatically.
```

To configure `daemon` mode, add a section to your `coroner.cf` file called `[daemon]`:

```bash
[daemon]
proj1.root=/home/jdoe/workspace/proj1
proj1.token=proj1
proj2.root=/home/jdoe/workspace/proj2
proj2.token=proj2
```

Each `root` setting specifies the path to a folder for `coroner` to watch, and the corresponding `token` specifies the friendly token name to use for the upload.

> In the example above, when a snapshot file is saved to `/home/jdoe/workspace/proj1`, `coroner daemon` will automatically execute the equivalent of `coroner put proj1 proj1 /home/jdoe/workspace/proj1/file.btt`.

Author: ```suggestion
In the example above, when a snapshot file is saved to `/home/jdoe/workspace/proj1`, `coroner daemon` will automatically run the equivalent of `coroner put proj1 proj1 /home/jdoe/workspace/proj1/file.btt`.
```
